lib/src/container/oci.rs

```diff
@@ -230,22 +230,23 @@ impl<'a> OciWriter<'a> {
         });
         let config_blob = write_json_blob(self.dir, &config)?;
 
-        let manifest_data = serde_json::json!({
-            "schemaVersion": default_schema_version(),
-            "config": {
-                "mediaType": OCI_TYPE_CONFIG_JSON,
-                "size": config_blob.size,
-                "digest": config_blob.digest_id(),
+        let mut annotations = BTreeMap::new();
+        annotations.extend(self.manifest_annotations.drain());
```

Comment:
Minor: this performs an implicit sorting step, iterating through an hashmap. I think it's a symptom which highlights an impedance mismatch in `OciWriter` definition. I'd suggest to switch that to directly use a `BTreeMap` or a `Vec`, so that are annotations are already in the expected order when we reach here.

Reply:
In practice here we use https://crates.io/crates/cjson which will order things.  But I can't imagine a case in which anything should actually depend on the ordering of the annotations.  So switched to `HashMap` for consistency.